lib/gui/app/models/storage.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2018 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const INDENTATION_SPACES = 2
+
+/**
+ * @summary Localstorage class and helper functions
+ * @class
+ * @public
+ */
+class Storage {
+  /**
+   * @function
+   * @public
+   *
+   * @param {String} superkey - superkey
+   *
+   * @example
+   * const potatoStorage = new Storage('potato')
+   */
+  constructor (superkey) {
+    this.superkey = superkey
+  }
+
+  /**
+   * @summary Get the whole object under the superkey
+   * @function
+   * @public
+   *
+   * @returns {Object}
+   *
+   * @example
+   * for (const key in potatoStorage.getAll()) {
+   *   console.log(key)
+   * }
+   */
+  getAll () {
+    try {
+      // JSON.parse(null) === null, so we fallback to {}
+      return JSON.parse(window.localStorage.getItem(this.superkey)) || {}
+    } catch (err) {
+      return {}
+    }
+  }
+
+  /**
+   * @summary Set the whole object under the superkey
+   * @function
+   * @public
+   *
+   * @param {Any} value - any valid JSON value
+   *
+   * @example
+   * potatoStorage.setAll({
+   *   location: 'somewhere',
+   *   freshness: 100,
+   *   edible: true
+   * })
+   */
+  setAll (value) {
+    window.localStorage.setItem(this.superkey, JSON.stringify(value, null, INDENTATION_SPACES))
```

Comment:
Analog to .getAll(), why does this not have error handling? There could certainly be the case of a value accidentally containing something that's not stringify-able, i.e. circular refs.

Reply:
Good call, but if we do try to stringify a circular reference, shouldn't we propagate the error and make it visible? We should probably log it with analytics too.